src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -311,6 +311,102 @@ void RegisterAppInterfaceRequest::Run() {
   SendRegisterAppInterfaceResponseToMobile();
 }
 
+void FillVRRelatedFields(smart_objects::SmartObject& response_params,
+                         const HMICapabilities& hmi_capabilities) {
+  response_params[strings::language] = hmi_capabilities.active_vr_language();
+  if (hmi_capabilities.vr_capabilities()) {
+    response_params[strings::vr_capabilities] =
+        *hmi_capabilities.vr_capabilities();
+  }
+}
+
+void FillVIRelatedFields(smart_objects::SmartObject& response_params,
+                         const HMICapabilities& hmi_capabilities) {
+  if (hmi_capabilities.vehicle_type()) {
+    response_params[hmi_response::vehicle_type] =
+        *hmi_capabilities.vehicle_type();
+  }
+}
+
+void FillUIRelatedFields(smart_objects::SmartObject& response_params,
+                         const HMICapabilities& hmi_capabilities) {
+    response_params[strings::hmi_display_language] =
+        hmi_capabilities.active_ui_language();
+    if (hmi_capabilities.display_capabilities()) {
+      response_params[hmi_response::display_capabilities] =
+          smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+      smart_objects::SmartObject& display_caps =
+          response_params[hmi_response::display_capabilities];
+
+      display_caps[hmi_response::display_type] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::display_type);
+
+      display_caps[hmi_response::text_fields] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::text_fields);
+
+      display_caps[hmi_response::image_fields] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::image_fields);
+
+      display_caps[hmi_response::media_clock_formats] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::media_clock_formats);
+
+      display_caps[hmi_response::templates_available] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::templates_available);
+
+      display_caps[hmi_response::screen_params] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::screen_params);
+
+      display_caps[hmi_response::num_custom_presets_available] =
+          hmi_capabilities.display_capabilities()->getElement(
+              hmi_response::num_custom_presets_available);
+
+      if (hmi_capabilities.display_capabilities()
+              ->getElement(hmi_response::image_capabilities)
+              .length() > 0) {
+        display_caps[hmi_response::graphic_supported] = true;
+      } else {
+        display_caps[hmi_response::graphic_supported] = false;
+      }
+
```

Comment:
@LuxoftAKutsan, isn't it better this way (instead if/else):
display_caps[hmi_response::graphic_supported] = 
0 < hmi_capabilities.display_capabilities()->getElement(hmi_response::image_capabilities).length();